File: 83.js
function backtrack(nums, i, path, res) {
    if (i === nums.length) {
        res.push([...path]);
        return;
    }

    backtrack(nums, i + 1, path, res);
    path.push(nums[i]);
    backtrack(nums, i + 1, path, res);
    path.pop();
}

const nums = [1, 2];
const res = [];
backtrack(nums, 0, [], res);

console.log(res);

File: 83.py
def backtrack(nums, i, path, result):
    if i == len(nums):
        result.append(path[:])
        return

    backtrack(nums, i + 1, path, result)
    path.append(nums[i])
    backtrack(nums, i + 1, path, result)
    path.pop()

nums = [1, 2]
result = []
backtrack(nums, 0, [], result)

for r in result:
    print(r)
